lib/royal_mail_api/xml_builder.rb

```diff
@@ -3,13 +3,21 @@
 class XmlBuilder
   attr_reader :request
 
+  SPECIAL_CHARACTER_MAP = {
+    '"' => "&quot;",
+    "&" => "&amp;",
+    "'" => "&apos;",
+    "<" => "&lt;",
+    ">" => "&gt;"
+  }
+
   def initialize(request, attrs={})
     @request = request
 
     # TODO change to avoid multi-thread mayhem
     attrs.each do |k,v|
       self.class.send(:attr_accessor, k)
```

Comment:
why not you use a hash with getter/setter instead of instance variables and this kind of hacks?